### PR TITLE
add a `fast` make target for kind-clustermesh

### PR DESCRIPTION
**Pull request by thorn3r**
The current make targets for making a pair of clustermesh'd kind clusters builds all container images and loads them on all nodes for both clusters. This is slow and quite resource intensive.

This commit adds a `kind-install-cilium-clustermesh-fast` target, which utilizes the existing `kind-image-fast` target to build and copy only the binaries.

The workflow for utilizing fast builds for clustermesh looks like this:

$ make kind-clustermesh
$ make kind-install-cilium-clustermesh-fast

And can be followed with any of the `kind-image-fast*` targets to re-build/copy the binaries.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -578,6 +578,34 @@ kind-install-cilium-clustermesh: kind-clustermesh-ready ## Install a local Ciliu
 	$(CILIUM_CLI) clustermesh status --context kind-clustermesh1 --wait
 	$(CILIUM_CLI) clustermesh status --context kind-clustermesh2 --wait
 
+kind-install-cilium-clustermesh-fast:kind-clustermesh-ready
```
Comment (qmonnet) [inline Makefile]: Missing space

```suggestion
kind-install-cilium-clustermesh-fast: kind-clustermesh-ready ## ...
```

It also misses the comment to briefly document the target so that the description shows up in `make help`.

More generally, should the `fast` target be documented somewhere?

Comment (qmonnet) [inline Makefile]: `kind-install-cilium-clustermesh-fast` should be added to `.PHONY` (the call to `KIND_ENV` takes care of it for `kind-install-cilium-clustermesh`).

File: Makefile [hunk 2]
```diff
@@ -578,6 +578,34 @@ kind-install-cilium-clustermesh: kind-clustermesh-ready ## Install a local Ciliu
...
+		--values=$(ROOT_DIR)/contrib/testing/kind-clustermesh2.yaml \
+		--values=$(ROOT_DIR)/contrib/testing/kind-fast.yaml \
+		--set=clustermesh.apiserver.kvstoremesh.enabled=$(ENABLE_KVSTOREMESH)
+
+	$(MAKE) kind-image-fast
+
+	@echo "  CONNECT the two clusters"
+	$(CILIUM_CLI) clustermesh connect --context kind-clustermesh1 --destination-context kind-clustermesh2
+	$(CILIUM_CLI) clustermesh status --context kind-clustermesh1 --wait
+	$(CILIUM_CLI) clustermesh status --context kind-clustermesh2 --wait
```
Comment (qmonnet) [inline Makefile]: I wouldn't mind some factoring between the two targets. Maybe the `CONNECT` part, for example, could be a separate target shared by the two (“slow” and fast) recipes? Not a strong requirement, though.

**Review by qmonnet — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (thorn3r) [reply Makefile]: Agreed that it seems less DRY than it could be. I spent some time trying to find an angle to just supply `FAST` as an env to avoid creating a new target, but that ended up being more complex and doesn't follow the existing convention with the other `fast` target.

I think splitting at least the connect target makes sense :+1:

**Review by qmonnet — APPROVED** [review]
Looks good to me, thanks for the update!

Now you've got a merge conflict though - Apologies, that's my PR that got merged just this morning, bad timing. Please port your changes to `Makefile.kind`.

**Review by qmonnet — APPROVED** [review]

File: Makefile.kind [hunk 3]
```diff
@@ -114,10 +121,33 @@ kind-install-cilium-clustermesh: kind-clustermesh-ready ## Install a local Ciliu
...
 		--set=clustermesh.apiserver.kvstoremesh.enabled=$(ENABLE_KVSTOREMESH)
 
-	@echo "  CONNECT the two clusters"
-	$(CILIUM_CLI) clustermesh connect --context kind-clustermesh1 --destination-context kind-clustermesh2
-	$(CILIUM_CLI) clustermesh status --context kind-clustermesh1 --wait
-	$(CILIUM_CLI) clustermesh status --context kind-clustermesh2 --wait
+	$(MAKE) kind-connect-clustermesh
+
+.PHONY: kind-install-cilium-clustermesh-fast
+kind-install-cilium-clustermesh-fast: kind-clustermesh-ready ## Install a local Cilium version into the ClusterMesh clusters and enable ClusterMesh.
```
Comment (glibsm) [inline Makefile.kind]: nit: comment is now wrong, it just loads the images not installs it

File: contrib/testing/kind-fast.yaml [hunk 4]
```diff
@@ -1,25 +1,25 @@
 extraVolumes:
-  - name: cilium-dbg-binary
```
Comment (glibsm) [inline contrib/testing/kind-fast.yaml:2]: nit: this is good cleanup, but technically should be a different commit

**Review by glibsm — APPROVED** [review]

File: Makefile.kind (hunk 3 above)
Comment (thorn3r) [reply Makefile.kind]: :thinking: It uses the CLI to install Cilium right after the image load, though

File: contrib/testing/kind-fast.yaml (hunk 4 above)
Comment (thorn3r) [reply contrib/testing/kind-fast.yaml:2]: agreed, i didn't initially intend to reformat the whole file but my editor got excited. Intention was just to add the needed volume and mount

**Comment by thorn3r** [comment]
CI is all green except `Conformance Ginkgo`, which is repeatedly hitting this flake: https://github.com/cilium/cilium/issues/30101

File: Makefile.kind (hunk 3 above)
Comment (glibsm) [reply Makefile.kind]: I was just trying to make the point that the comment is the same, I didn't do a great job phrasing that. If we're ok with them both being the same no problem

Comment (thorn3r) [reply Makefile.kind]: oh yes you're right, updated the comment (and the one for `kind-install-cilium-fast` as well, in a separate commit)
